include/kernel.h

```diff
@@ -0,0 +1,531 @@
+ * so that it's possible for threads to wait for specific interrupts and
+ * act after them.
+ *
+ * It is also possible to attach a mailbox to an event using #mbox_attach_event.
+ * This is useful if a thread needs to wait for multiple events.
+ */
+typedef struct kevent_s
+{
+	/** Mailboxes attached to this event */
+	kmbox_t *mboxes[MAX_MAILBOXES_PER_EVENT];
+} kevent_t;
+
+
+/** @brief Event generated when a SP interrupt is triggered */
+extern kevent_t KEVENT_IRQ_SP;
+/** @brief Event generated when a SI interrupt is triggered */
+extern kevent_t KEVENT_IRQ_SI;
+/** @brief Event generated when a AI interrupt is triggered */
+extern kevent_t KEVENT_IRQ_AI;
+/** @brief Event generated when a VI interrupt is triggered */
+extern kevent_t KEVENT_IRQ_VI;
+/** @brief Event generated when a PI interrupt is triggered */
+extern kevent_t KEVENT_IRQ_PI;
+/** @brief Event generated when a DP interrupt is triggered */
+extern kevent_t KEVENT_IRQ_DP;
+
+
+/** 
+ * @brief Initialize the multi-threading kernel
+ *
+ * The current execution context becomes the main thread of the
+ * program, with priority set to 0 (you can change priority
+ * to the main thread as well using #thread_set_pri).
+ *
+ * The main thread uses the original stack allocated for the
+ * whole process, so it is technically unbounded (or limited by
+ * heap size).
+ *
+ * @return a reference to the main thread.
+ *
+ */
+kthread_t* kernel_init(void);
+
+/** 
+ * @brief Shutdown the multi-threading kernel
+ *
+ * This function is mostly useful for testing purposes.
+ * Since the kernel does not keep track of all created threads,
+ * this function should be called only when all created threads
+ * are exited or have been killed.
+ */
+void kernel_close(void);
+
+/**
+ * @brief Create a new thread
+ *
+ * Create a new thread, with a specified stack size and priority.
+ * The thread is immediately made ready after creation, so if it has
+ * a priority higher than or equal to the current thread, it will be
+ * scheduled immediately, before thread_new returns.
+ *
+ * The thread will begin execution from the specified entry point
+ * function. If the function ever returns, the thread is automatically
+ * killed.
+ *
+ * @param[in] name
+ *            Name of the thread (for debugging purposes)
+ * @param[in] stack_size
+ *            Size of the stack in bytes. Minimum suggested size
+ *            is 2048.
+ * @param[in] pri
+ *            Priority of the thread. Negative values are not allowed.
+ *            Higher number means higher priority.
+ * @param[in] user_entry
+ *            Entry point of the thread. This is the function that will
+ *            be called when the thread begins execution. If this function
+ *            ever returns, the thread is automatically killed.
+ * @param[in] user_data
+ *            Argument that will be passed to the entry point of the thread.
+ *
+ * @return    A reference to the thread. It is not necessary to store
```

Comment:
```suggestion
 * @return    A pointer to the new thread. It is not necessary to store
```

Just another naming nitpick.

Reply:
Applied.